File: packages/engine/lib/src/planner/load_budget_config.dart
import 'package:meta/meta.dart';

import '../models/enums.dart';

/// The slice of `content/planer.yaml` the load budget needs (sections
/// `belastbarkeit_pro_tag`, `einschraenkung_deckel` and `erholungsbedarf`).
/// Passed in, not imported (CLAUDE.md, rule 10).
@immutable
class LoadBudgetConfig {
  const LoadBudgetConfig({
    required this.capacityPerDay,
    required this.restrictionCap,
    required this.recoveryNeedMediumFrom,
    required this.recoveryNeedHighFrom,
  });

  final Map<LifeStage, double> capacityPerDay;

  /// Only restrictions that cap capacity have an entry here — others (e.g.
  /// `jointIssues`, `senior`) act elsewhere in the planner (filtering).
  final Map<Restriction, double> restrictionCap;

  final double recoveryNeedMediumFrom;
  final double recoveryNeedHighFrom;
}

File: infra/supabase/functions/_shared/planner/steps/load_budget_config.ts
import type { LifeStage, Restriction } from '../models/enums.ts';

/**
 * The slice of `content/planer.yaml` the load budget needs (sections
 * `belastbarkeit_pro_tag`, `einschraenkung_deckel` and `erholungsbedarf`).
 * Passed in, not imported (CLAUDE.md, rule 10).
 */
export interface LoadBudgetConfig {
  readonly capacityPerDay: ReadonlyMap<LifeStage, number>;

  /**
   * Only restrictions that cap capacity have an entry here — others (e.g.
   * `jointIssues`, `senior`) act elsewhere in the planner (filtering).
   */
  readonly restrictionCap: ReadonlyMap<Restriction, number>;

  readonly recoveryNeedMediumFrom: number;
  readonly recoveryNeedHighFrom: number;
}
